hcl2template/types.packer_config.go

```diff
@@ -40,18 +40,20 @@ type PackerConfig struct {
 	InputVariables Variables
 	LocalVariables Variables
 
+	DataSources DataSources
+
 	LocalBlocks []*LocalBlock
 
 	ValidationOptions
 
 	// Builds is the list of Build blocks defined in the config files.
 	Builds Builds
 
-	builderSchemas packer.BuilderStore
-
-	provisionersSchemas packer.ProvisionerStore
-
+	datasources           map[string]cty.Value
```

Comment:
I think this should not be in here but rather returned by the function creating it. The `PackerConfig` represents a loaded hcl config in the hcl sense. Where execution leads to values.

Reply:
We use the same value more than once, for locals and for builds. We need to keep these save somewhere to avoid executing again and again. I'll try to think of a solution that keeps this behavior.

Reply:
Gotcha ! Yes We need to have some sort of more explicit 'storage' here. 🙂